opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/_load.py

```diff
@@ -31,9 +30,32 @@
 _logger = getLogger(__name__)
 
 
+class _EntryPointDistFinder:
+    def __int__(self):
+        self._mapping = None
```

Comment:
Consider making this a `functools.cached_property`?

Reply:
Something like this?:

```
    @cached_property
    def dist_for(self, entry_point: EntryPoint):
        dist = getattr(entry_point, "dist", None)
        if dist:
            return dist

        if self._mapping is None:
            self._mapping = {
                self._key_for(ep): dist
                for ep in dist.entry_points
                for dist in distributions()
            }

        return self._mapping.get(self._key_for(entry_point))
```



This would always return the same dist, right?

Reply:
I meant the _mapping attribute within _EntrypointDistFinder - it'll do exactly what you're already doing and only compute _mapping once

Reply:
Okay yes, I agree

Reply:
@Arnatious isn't that missing the dist parameter though?

Reply:
I've tried this (with the two for lines swapped) but it looks like the code is not used with our current tests, see 
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2871/files#r1759148502

So the question is: do we really need this cache?

Reply:
Looks like according to coverage everything we test has a dist attribute, and therefore exits before the call to ._mapping

Reply:
Granted, not sure that's necessarily something _this_ pr needs to fix.